pkg/commands/archive.go

```diff
@@ -0,0 +1,136 @@
+package commands
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+
+	build "github.com/Azure/acr-builder/pkg"
+	"github.com/Azure/acr-builder/pkg/constants"
+)
+
+// ArchiveSource defines source in the form of an archive file
+// Currently we only support tar.gz
+type ArchiveSource struct {
+	url       string
+	targetDir string
+	tracker   *DirectoryTracker
+}
+
+// NewArchiveSource creates a new archive source
+func NewArchiveSource(url string, targetDir string) build.Source {
+	return &ArchiveSource{url: url, targetDir: targetDir}
+}
+
+// Obtain downloads and extract the source
+func (s *ArchiveSource) Obtain(runner build.Runner) error {
+	response, err := http.Get(s.url)
```

Comment:
>http.Get(s.url) [](start = 18, length = 15)

Would be good to enable timing info and output steps of download and extraction with timing here.

Reply:
https://github.com/Azure/acr-builder/issues/22